doc/contributing/index.rst

```diff
@@ -0,0 +1,250 @@
+
+.. code-block:: python
+
+    pip install pytest
+
+You can use `pytest` on your python code via the following instructions:
+
+.. code-block:: python
+
+    pytest name_of_your_script.py
+
+2. Using Azure (azure-pipelines.yml) and giotto-tda's CI scripts.
+
+
+*************
+Documentation
+*************
+
+Description of the infrastructure
+=================================
+
+The documentation is hosted on github-pages (on `https://giotto-ai.github.io/gtda-docs/ <https://giotto-ai.github.io/gtda-docs/>`_).
+It has 3 main components:
+
+- API: auto-generated python documentation (sphinx),
+- contribution guidelines, README, theory page etc,
+- collection of notebook examples (from ../examples, converted to `.py` scripts and executed).
+
+All of the 3 components are `html` pages, generated as described in :ref:`How to build <build_docs>`.
+The generated pages have to be committed to a local clone of the
+`github repository containing the documentation <https://github.com/giotto-ai/gtda-docs>`_,
+which also contains documentation for previous versions of the project.
+
+Generating documentation has been automated and is now part of the Azure CI. Each merge to master generates a new documentation.
+
+How to build
+============
+
+.. _build_docs:
+
+Basic instructions
+------------------
+
+Create an environment using the ???? file and activate it. We assume that the `giotto-tda` and `gtda-docs` repositories
+are cloned in `/path/to/giotto-tda/` and `path/to/gtda-docs/`.
+
+.. code-block:: bash
+
+   cd path/to/giotto-tda/
+   cd doc/
+
+Set the environment variable `GITDIR` to `path/to/gtda-docs/`, either directly in the console or editing the `Makefile`:
+
+.. code-block:: bash
+
+   GITDIR = path/to/gtda-docs/
+
+Then, generating the documentation, copying it to the `gtda-docs` repository and committing is done with:
+
+.. code-block:: bash
+
+   make all_gh
+   make gh-commit
+
+Note that the success of the `gh-commit` target depends on whether you have writing rights to the `gtda-docs` repository.
+
+Step-by-step instructions
+-------------------------
+
+We describe in more details the targets in the Makefile, following the flow described above.
+
+Clean
+~~~~~
+
+The `all-gh` target starts with `clean-gh`` which uses the underlying sphinx `clean` target to remove files from the `build` directory.
+Similarly, we remove `theory/glossary.rst`, which might have been geenerated by a previous call to `make all-gh`.
+
+Theory
+~~~~~~
+
+**This step requires `pandoc` and `pandoc-citeproc` as additional dependencies.**
+
+The glossary `theory/glossary.tex` contains the mathematical definitions of the functionalities
+and the terms used throughout the documentation.
+
+The document, along with the bibliography, is converted to an `.rst`, using `pandoc`.
+It is included in the main `toctree` of the documentation.
+The only purpose and use of `theory/before_glossary.rst` is to have the reference/hyperlink.
```

Comment:
Any reason for this naming? Could there be a more revealing alternative?

Also, that file seems to only contain the line
```
.. _glossary_ref:
```
But it's unclear to me where this is ever used.

Reply:
The file is used, but i don't think i ever use `glossary_ref` - only references to the actual entries.  Probably a legacy of when i had a messy toctree, since otherwise unused. 
Deleting